pkg/controller/direct/bigquerydataset/dataset_controller.go

```diff
@@ -99,11 +108,12 @@ func (m *model) AdapterForURL(ctx context.Context, url string) (directbase.Adapt
 }
 
 type Adapter struct {
-	id         *krm.BigQueryDatasetRef
-	gcpService *api.Service
-	desired    *krm.BigQueryDataset
-	actual     *api.Dataset
-	reader     client.Reader
+	id                         *krm.BigQueryDatasetRef
+	gcpService                 *bigquery.Client
+	desired                    *krm.BigQueryDataset
+	actual                     *bigquery.DatasetMetadata
+	reader                     client.Reader
+	hasUpdateIsCaseInsensitive bool
```

Comment:
Not sure I follow, what does `hasUpdateIsCaseInsensitive` do?

Reply:
In go, explicitly set a bool value to false and not setting it (it defaults to false), there is not a method to tell if the difference. I added it here to tell if the update has explicitly set the value to false, therefore I should update this field. Or it only defaults to false, where I should not be updating it.